src/app/clusters/identify-server/tests/TestIdentifyCluster.cpp

```diff
@@ -0,0 +1,573 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "TestIdentifyClusterHelpers.h"
+#include <app/clusters/identify-server/IdentifyCluster.h>
+#include <pw_unit_test/framework.h>
+
+#include <app/data-model-provider/tests/ReadTesting.h>
+#include <app/data-model-provider/tests/WriteTesting.h>
+#include <app/server-cluster/AttributeListBuilder.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <app/server-cluster/testing/TestServerClusterContext.h>
+#include <clusters/Identify/Attributes.h>
+#include <clusters/Identify/Commands.h>
+#include <clusters/Identify/Metadata.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::Identify;
+using namespace chip::app::Clusters::Identify::Attributes;
+
+namespace {
+
+bool onIdentifyStartCalled    = false;
```

Comment:
using globals feels like a bit like it could cause ordering issues especially if some test fails. I don't think this is the case here as you are resetting them in startup, however wondering what would prevent us from building some RAII on this. And the reasoning seems to be the `on*` methods that just take in the cluster. If we would allow a `intptr_t` argument like generic C callbacks, we would allow a context or, maybe even better, if we would allow a Delegate pattern we could subclass it and make it work.

So could we:
  - Make the callbacks be actually a pure virtual class of IdentifyDelegate that has methods and RAAI this?
  - or (a bit uglier) make the callbacks take a intptr_t and still RAII this with propper casts?